### PR TITLE
Allow a supercommand to implement extra global flags

### PR DESCRIPTION
**Pull request by rogpeppe**
This means that a command can have flags that are
common to all subcommands and can be specified
before the subcommand name.

### CODE REVIEW

File: supercommand.go [hunk 1]
```diff
@@ -71,17 +76,25 @@ type SuperCommandParams struct {
...
 	command := &SuperCommand{
-		Name:                params.Name,
-		Purpose:             params.Purpose,
-		Doc:                 params.Doc,
-		Log:                 params.Log,
+		Name:    params.Name,
+		Purpose: params.Purpose,
+		Doc:     params.Doc,
+		Log:     params.Log,
+		Aliases: params.Aliases,
```
Comment (nskaggs) [inline supercommand.go]: Any reason why this was moved?

File: supercommand.go [hunk 2]
```diff
@@ -335,13 +348,16 @@ func (c *SuperCommand) SetCommonFlags(f *gnuflag.FlagSet) {
...
+		c.globalFlags.AddFlags(f)
+	}
 	f.BoolVar(&c.showHelp, "h", false, helpPurpose)
 	f.BoolVar(&c.showHelp, "help", false, "")
 	// In the case where we are providing the basis for a plugin,
 	// plugins are required to support the --description argument.
 	// The Purpose attribute will be printed (if defined), allowing
 	// plugins to provide a sensible line of text for 'juju help plugins'.
-	f.BoolVar(&c.showDescription, "description", false, "")
+	f.BoolVar(&c.showDescription, "description", false, "Show short description of plugin, if any")
```
Comment (nskaggs) [inline supercommand.go]: Why add the usage string here given it's not previously been defined.

File: supercommand.go (hunk 1 above)
Comment (rogpeppe) [reply supercommand.go]: So that it fits with all the other exported fields.

File: supercommand.go (hunk 2 above)
Comment (rogpeppe) [reply supercommand.go]: Because otherwise you get an awkward blank line in the help output. If we're going to define a flag, we should describe it.

Comment (rogpeppe) [reply supercommand.go]: (FWIW I don't really see why the flag is exported for all commands, but if it's going to be there, it should have some docs)

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
No bot enabled, so landing this manually.
Tests pass.

**Pull request merged**